client/login.go

```diff
@@ -79,29 +80,53 @@ func (client *Client) LoggedIn() bool {
 }
 
 func storeAuthDataFilename() string {
-	authFilename := os.Getenv("SNAPPY_STORE_AUTH_DATA_FILENAME")
-	if authFilename != "" {
-		return authFilename
-	}
-	homeDir, err := osutil.CurrentHomeDir()
+	real, err := osutil.RealUser()
 	if err != nil {
 		panic(err)
 	}
+
+	return storeAuthDataFilenameFromHome(real.HomeDir)
+}
+
+const authFileEnvKey = "SNAPPY_STORE_AUTH_DATA_FILENAME"
+
+func storeAuthDataFilenameFromHome(homeDir string) string {
+	if fn := os.Getenv(authFileEnvKey); fn != "" {
```

Comment:
Okay, can we at least merge them back into a single function, to avoid some of the complexity and the improper behavior of "from home" not being from home?
```
func storeAuthDataFilename(homeDir string) string {
```